zcash_primitives/src/transaction/blake2b_256.rs

```diff
@@ -0,0 +1,31 @@
+use blake2b_simd::{Hash, Params, State};
+use std::io::{self, Write};
+
+/// Abstraction over a writer which BLAKE2B-256-hashes the data being read.
+pub struct HashWriter {
```

Comment:
This should not be necessary, as `blake2b_simd` implements `std::io::Write` on its hashers (when the `std` feature flag is enabled, which it is by default).